src/qibolab/qubits.py

```diff
@@ -90,10 +90,24 @@ class Qubit:
     feedback: Optional[Channel] = None
     twpa: Optional[Channel] = None
     drive: Optional[Channel] = None
-    flux: Optional[Channel] = None
+    _flux: Optional[Channel] = None
 
     native_gates: SingleQubitNatives = field(default_factory=SingleQubitNatives)
 
+    def __post_init__(self):
+        if self.flux is not None and self.sweetspot != 0:
+            self.flux.offset = self.sweetspot
+
+    @property
+    def flux(self):
+        return self._flux
+
+    @flux.setter
+    def flux(self, channel):
+        if self.sweetspot != 0:
+            channel.offset = self.sweetspot
+        self._flux = channel
```

Comment:
Do you believe this is actually required?

Here we are duplicating the information, in channel and qubit, and then we have to increase the code complexity to keep them in sync.

In general, it is good to have a single source of truth, and always read from there (currently, if the `.sweetspot` is updated afterward, `._flux` get out of sync, unless properly corrected).

The problem is that the qubit depends on the channel, and not the opposite.
However, the qubit is the parameters' container, and the channel (and especially the port) is the one acting to send pulses.

So, eventually you need to play a pulse on a qubit, and this information goes down to the instruments (the `.play()` or `.sweep()` methods of the controllers, that we may eventually unify https://github.com/qiboteam/qibolab/blob/d881d1f7ebccaabf488224f26d69e35f8d279418/src/qibolab/instruments/qm/driver.py#L111-L112).
However, the instrument has no access to the channels, that's why is provided through `Qubit`. But in principle, we could even invert, and play the pulse on a `Channel` and a `QubitId` (even a `Port` is not unique, in case of multiplexing).

Reply:
I'm not sure that inverting the dependency is simple and the most effective strategy (esp. in the short term).
But I just see the duplication, and that you're trying to cope for it with syncing. And I know this is hard to maintain, in general.

Reply:
I also don't fully like this. In terms of this PR I did it to keep the behavior the same after the removal of `platform.setup()`.

In general `channel.offset` can be different than `qubit.sweetspot`, for example if `channel` is not a flux channel or if we don't want to operate a qubit at its sweetspot. So it is not double source of truth, we are just assuming that we are operating all qubits at their sweetspots and we are setting the flux offset accordingly.

Other potential solutions I can think of are:
1. Drop `offset` from channels and modify drivers to use `qubit.sweetspot` as the offset of flux. This has many downsides as `offset` may be useful for non-flux channels and it will not be possible to operate qubits not at their sweetspot.
2. Drop `qubit.sweetspot` and just put the value on the `offset` directly as an instrument setting. Main downside of this is "political", people doing calibration may find it weird and the format of the runcards will change a bit so it should be communicated. Other than that that's not necessarily bad solution.
3. Keep both but the user has to set both manually. More flexible but will lead to repetition in the runcards, as the same value will be set twice in most cases. This is similar to the qubit frequency which is set as `qubit.drive_frequency` and also as the frequency of the native RX pulse.

Reply:
> In general `channel.offset` can be different than `qubit.sweetspot`, for example if `channel` is not a flux channel or if we don't want to operate a qubit at its sweetspot. So it is not double source of truth, we are just assuming that we are operating all qubits at their sweetspots and we are setting the flux offset accordingly.

Even @hay-k mentioned this to me, and he mentioned that also @aorgazf asked for this *feature* (that in practice consists in doing something less).
But I agree that for this PR might be better to keep the same behavior (just because the PR is about something else).

Solution 1. I don't like, because as you pointed out they could be two different objects, and because we don't want to repeat over and over some decision in all the drivers.
Every entity should be represented, and we should make proper use of them. If we want to change the offset, let's do it at higher level.

As you said yourself, solution 2. is better software-wise. But if we calibrate the sweetspot, it has to be recorded somewhere.

Solution 3. is close to what we have, and it's the one I'd keep. I'd simply drop the automated sync at this point.
We can have a mechanism (a method? a parameter in other method?) to do that (set the `channel.offset` out of the `qubit.sweetspot`) if it's a convenient shortcut for reading from the one side and setting to the other. Or otherwise do it manually. But the decision should always be explicit (unless we have an offset setting stage, and we just keep sweetspot as the default - but this stage should be easily accessible by the user).

Reply:
> But I agree that for this PR might be better to keep the same behavior (just because the PR is about something else).

Indeed, I opened an issue (#741) from this discussion. Let's come back to it in the next (or next next) milestone.